### PR TITLE
intake to transfer

### PR DESCRIPTION
**Pull request by yutytuty**

### CODE REVIEW

File: src/main/java/frc/robot/Subsystems/IntakeToTransfer.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    private static IntakeToTransfer instance;
+
+    public static IntakeToTransfer getInstance() {
+        if (instance == null) {
+            instance = new IntakeToTransfer(new WPI_TalonSRX(RobotMap.CAN.INTAKE_TO_TRANSFER_TALON));
+        }
+        return instance;
+    }
+
+    public IntakeToTransfer(WPI_TalonSRX talon) {
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/IntakeToTransfer.java:17]: private?

File: src/main/java/frc/robot/Subsystems/IntakeToTransfer.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+import frc.robot.RobotMap;
+
+public class IntakeToTransfer extends MotoredGenericSubsystem {
```
Comment (tal-sitton) [inline src/main/java/frc/robot/Subsystems/IntakeToTransfer.java:7]: where limit?

**Review by tal-sitton — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/Subsystems/IntakeToTransfer.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        this.transfer = Transfer.getInstance();
+    }
+
+    public boolean getLimit() {
+        return limit.get();
+    }
+
+    @Override
+    public boolean canMove(double speed) {
+        return !transfer.getStrapEntranceSensor() || !limit.get();
```
Comment (yutytuty) [inline src/main/java/frc/robot/Subsystems/IntakeToTransfer.java:34]: reformat

File: src/main/java/frc/robot/RobotMap.java [hunk 4]
```diff
@@ -1,4 +1,10 @@
 package frc.robot;
 
 public class RobotMap {
+    public interface CAN {
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/RobotMap.java:4]: blank line at the start of the inteface

File: src/main/java/frc/robot/RobotMap.java [hunk 5]
```diff
@@ -1,4 +1,10 @@
 package frc.robot;
 
 public class RobotMap {
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/RobotMap.java:3]: blank line at the start of the class

File: src/main/java/frc/robot/RobotMap.java [hunk 6]
```diff
@@ -1,4 +1,10 @@
...
+    public interface CAN {
+        int INTAKE_TO_TRANSFER_TALON = -1;
+    }
+    public interface DIO {
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/RobotMap.java:7]: add blank line between intefaces

File: src/main/java/frc/robot/RobotMap.java [hunk 7]
```diff
@@ -1,4 +1,10 @@
...
+        int INTAKE_TO_TRANSFER_LIMIT = -1;
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/RobotMap.java:8]: blank line at the start of the interface

File: src/main/java/frc/robot/Subsystems/IntakeToTransfer.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.robot.RobotMap;
+
+public class IntakeToTransfer extends MotoredGenericSubsystem {
+    private static IntakeToTransfer instance;
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/Subsystems/IntakeToTransfer.java:10]: blank line at the start of the class

File: src/main/java/frc/robot/Subsystems/IntakeToTransfer.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
+package frc.robot.Subsystems;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import com.spikes2212.command.genericsubsystem.MotoredGenericSubsystem;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.robot.RobotMap;
+
+public class IntakeToTransfer extends MotoredGenericSubsystem {
```
Comment (OfriRosenbaum) [inline src/main/java/frc/robot/Subsystems/IntakeToTransfer.java:9]: what is this class?

**Review by OfriRosenbaum — CHANGES_REQUESTED** [review]

**Review by OfriRosenbaum — APPROVED** [review]

**Review by tal-sitton — APPROVED** [review]

**Pull request merged**